### PR TITLE
Add SciPy builder.

### PR DESCRIPTION
**Pull request by arielshaqed**
Usage:

1. Go to your Binaris directory.
2. Add `requirements.txt` (otherwise it uses its default).
3. Run `/path/to/bob/scipy/build.sh`.
4. `bn deploy functionName`.

Tested by building and deploying.

### CODE REVIEW

File: scipy/build.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+$DOCKER build --tag=pipper $dir # Keep going even on failure!
+if [ -f ./requirements.txt ]; then
+  volume="-v $PWD/requirements.txt:/function/requirements.txt"
+else
+  volume=''
+fi
+$DOCKER run $volume pipper:latest
+build_status=$?
+$DOCKER run pipper:latest bash -c 'tar -C /cache -cz .' | tar -x --keep-newer -C .binaris/.pycache/
```
Comment (bergundy) [inline scipy/build.sh:24]: --rm

File: scipy/build.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+$DOCKER run pipper:latest bash -c 'tar -C /function -cz .' | tar -x --keep-newer -C dist/
```
Comment (bergundy) [inline scipy/build.sh:25]: --rm

File: scipy/build.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
```
Comment (bergundy) [inline scipy/build.sh:2]: I'd feel much more comfortable if you add `set -euo pipefail`.
Next time someone edits this file they'll assume the new command they just added will fail the script...

File: scipy/docker/install.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+set -x
+
+# Installation script for requirements.txt to get around weird
+# installers for scipy and scikit-learn.  Assumes scipy dependencies
+# are already installed.
+
+PIP='pip install -t ./ --install-option=--prefix='
+
+export PYTHONPATH=$PWD          # So scikit-learn installer sees
+                                # installations to this dir.
```
Comment (bergundy) [inline scipy/docker/install.sh:12]: `sees installations to`?

File: scipy/docker/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#aerospike==2.1.2
```
Comment (bergundy) [inline scipy/docker/requirements.txt:1]: Why are these requirements here?

File: scipy/docker/install.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set +e
```
Comment (bergundy) [inline scipy/docker/install.sh:2]: Did you mean `set -e`?
Also consider adding `u` and `o pipefail`

**Review by bergundy — CHANGES_REQUESTED** [review]

File: scipy/build.sh (hunk 3 above)
Comment (arielshaqed) [reply scipy/build.sh:2]: Done!  This also required some fancy `||...&&` work.

File: scipy/build.sh (hunk 1 above)
Comment (arielshaqed) [reply scipy/build.sh:24]: Done (also below).

File: scipy/docker/install.sh (hunk 6 above)
Comment (arielshaqed) [reply scipy/docker/install.sh:2]: Done!

File: scipy/docker/install.sh (hunk 4 above)
Comment (arielshaqed) [reply scipy/docker/install.sh:12]: Improved comment.

File: scipy/docker/requirements.txt (hunk 5 above)
Comment (arielshaqed) [reply scipy/docker/requirements.txt:1]: They're a useful basis.  Shall I change to just numpy, scipy, scikit-learn?

**Review by rylandg — APPROVED** [review]

**Comment by arielshaqed** [comment]
Pulling since @bergundy is asleep.

Thanks!

**Pull request merged**